File: java/src/test/java/problems101_200/ValidPalindromeTest.java
package problems101_200;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class ValidPalindromeTest extends SampledTest {
    @Test
    public void test() {
        for (ValidPalindrome solution : ValidPalindrome.values()) {
            for (ValidPalindromeSample sample : getSamples(ValidPalindromeSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.isPalindrome(sample.input.s))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems101_200/ValidPalindromeTest.kt
package problems101_200

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import kotlin.test.Test

class ValidPalindromeTest : SampledTest() {
    @Test
    fun test() =
        ValidPalindrome.entries.forEach { solution ->
            getSamples(Array<ValidPalindromeSample>::class.java)
                .forEach { (input, output, message) ->
                    assertWithMessage(message)
                        .that(solution.isPalindrome(input.s))
                        .isEqualTo(output)
                }
        }
}
